tools/advisor/advisor/db_config_optimizer.py

```diff
@@ -0,0 +1,414 @@
+            new_value = int(old_value * mul + add)
+        return new_value
+
+    # TODO(poojam23): try other algorithms for improving the database
+    # configuration
+    # currently, this method picks ONE option at random, and makes
+    # appropriate changes to that option for ALL the column families.
+    @staticmethod
+    def improve_db_config(current_config, guidelines):
+        option = random.choice(list(guidelines.keys()))
+        better_config = {option: {}}
+        for action in guidelines[option]:
+            suggested_values = (
+                guidelines[option][action][ConfigOptimizer.SUGG_VAL]
+            )
+            for col_fam in guidelines[option][action][ConfigOptimizer.SCOPE]:
+                old_value = None
+                if (
+                    option in current_config and
+                    col_fam in current_config[option]
+                ):
+                    old_value = float(current_config[option][col_fam])
+                try:
+                    new_value = ConfigOptimizer.apply_action_on_value(
+                        old_value, action, suggested_values
+                    )
+                    better_config[option][col_fam] = new_value
+                except AssertionError:
+                    print(
+                        'WARNING(ConfigOptimizer): provide suggested_values ' +
+                        'for ' + option
+                    )
+        return better_config
+
+    @staticmethod
+    def improve_db_config_v2(options, rule, suggestions_dict):
+        '''
+        this method takes ONE 'rule' and applies all its suggestions on the
+        appropriate options whereas
+        'improve_db_config' picks ONE option to update and applies all
+        suggestions that were triggered for that option
+        '''
+        # get the current configuration
+        required_options = []
+        rule_suggestions = []
+        for sugg_name in rule.get_suggestions():
+            option = suggestions_dict[sugg_name].option
+            action = suggestions_dict[sugg_name].action
+            if not (option and action):  # suggestion not in required format
+                continue
+            required_options.append(option)
+            rule_suggestions.append(suggestions_dict[sugg_name])
+        current_config = options.get_options(required_options)
+        # Create the updated configuration from the rule's suggestions
+        updated_config = {}
+        for sugg in rule_suggestions:
+            # case: when the option is not present in the current configuration
+            if sugg.option not in current_config:
+                try:
+                    new_value = ConfigOptimizer.apply_action_on_value(
+                        None, sugg.action, sugg.suggested_values
+                    )
+                    if sugg.option not in updated_config:
+                        updated_config[sugg.option] = {}
+                    if '.' not in sugg.option:
```

Comment:
can you add inline comment to explain the significance of this check?

Reply:
As discussed offline lets clarify it with more inline comments when we parse the rules as well.